### PR TITLE
fix: opening file starting with a plus

### LINKED ISSUE #2502: Filenames beginning with a +
### Description

Telescope can't open a file with a leading + symbol since + has a special meaning in vim.

### Neovim version

```markdown
NVIM v0.9.0
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/gcc -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wvla -Wdouble-promotion -Wmissing-noreturn -Wmis
sing-format-attribute -Wmissing-prototypes -fno-common -Wno-unused-result -Wimplicit-fallthrough -fdiagnostics-color=auto -fstack-protector-strong -DUNIT_TESTING -DINCLUDE_GENERATED_DECLAR
ATIONS -D_GNU_SOURCE -I/usr/include/luajit-2.1 -I/usr/include -I/usr/include/luajit-2.1 -I/builddir/build/BUILD/neovim-0.9.0/redhat-linux-build/src/nvim/auto -I/builddir/build/BUILD/neovim
-0.9.0/redhat-linux-build/include -I/builddir/build/BUILD/neovim-0.9.0/redhat-linux-build/cmake.config -I/builddir/build/BUILD/neovim-0.9.0/src -I/usr/include -I/usr/include -I/usr/include
 -I/usr/include -I/usr/include -I/usr/include -I/usr/include

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

### Operating system and version

Linux 6.2.14

### Telescope version / branch / rev

telescope 0.1.1

### checkhealth telescope

```markdown
telescope: require("telescope.health").check()

Checking for required plugins ~
- OK plenary installed.
- OK nvim-treesitter installed.

Checking external dependencies ~
- OK rg: found ripgrep 13.0.0
- OK fd: found fd 8.7.0

===== Installed extensions ===== ~

Telescope Extension: `fzf` ~
- OK lib working as expected
- OK file_sorter correctly configured
- OK generic_sorter correctly configured

Telescope Extension: `media_files` ~
- No healthcheck provided
```

### Steps to reproduce

1. touch +page.svelte
2. vim
3. :Telescope find_files
4. Select the +page.svelte
5. Doesn't open!!

### Expected behavior

_No response_

### Actual behavior

File doesn't open.

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      {
        'nvim-telescope/telescope.nvim',
        requires = {
          'nvim-lua/plenary.nvim',
          { 'nvim-telescope/telescope-fzf-native.nvim', run = 'make' },
        },
      },
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
      display = { non_interactive = true },
    },
  }
end
_G.load_config = function()
  require('telescope').setup()
  require('telescope').load_extension('fzf')
  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Telescope and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
load_plugins()
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

### PR DESCRIPTION
**Pull request by Conni2461**
fix #2502

### CODE REVIEW

File: lua/telescope/actions/set.lua [hunk 1]
```diff
@@ -165,6 +165,9 @@ action_set.edit = function(prompt_bufnr, command)
     -- prevents restarting lsp server
     if vim.api.nvim_buf_get_name(0) ~= filename or command ~= "edit" then
       filename = Path:new(vim.fn.fnameescape(filename)):normalize(vim.loop.cwd())
+      if filename:sub(1, 1) == "+" then
+        filename = "\\" .. filename
+      end
```
[the remark this answers is not in the document]
Comment (jamestrew) [reply lua/telescope/actions/set.lua:170]: > so fnameescape breaks $types.d.ts in cwd e.g.

really? I'm not experiencing this
touch "\$types.d.ts" followed by 
```
:lua vim.cmd(string.format("e %s", vim.fn.fnameescape("$types.d.ts")))
```

[the remark this answers is not in the document]
Comment (Conni2461) [reply lua/telescope/actions/set.lua:170]: seems like the issue is that we do the fnameescape twice, once outside and once inside Path, thats why it wasnt working, when i just tested it

**Comment by Conni2461** [comment]
lets try it, but i am not gonna backport it yet, gonna wait one day

**Comment by Conni2461** [comment]
I can't seem to reproduce any errors/files that dont seem to open correctly.

Thanks for the help :)

**Pull request merged**